aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -194,9 +195,28 @@ private void openStream() throws IOException {
     }
   }
 
-  private void closeStream() throws IOException {
+  private void closeStream() {
     if (stream != null) {
-      stream.close();
+      // if we aren't at the end of the stream, and the stream is abortable, then
+      // call abort() so we don't read the remaining data with the Apache HTTP client
+      abortStream();
+      try {
+        stream.close();
+      } catch (Exception e) {
+        // close quietly, closing an aborted stream will throw a content length
```

Comment:
Sorry I guess I did not express it clearly. What I meant was that, can we check for the exception that if it is a content length check exception then we can skip, otherwise we log a warning?

Reply:
I see. `ContentLengthInputStream` is an Apache HTTP class, so referring to that will tie S3InputStream to that client.

Reply:
I could check the name of the class, but that seems like a little bit of a hack.

Reply:
Ah okay, what does the exception look like? I though there are some error code that we can get out of the exception that would imply this specific error. Is that not there?

Reply:
[This](https://github.com/apache/httpcomponents-core/blob/2135596b1319910d56efb148566ecbbec0884638/httpcore/src/main/java/org/apache/http/impl/io/ContentLengthInputStream.java#L147)  is what gets thrown. Calling close on the aborted stream will attempt a read which then throws that.

Reply:
Checking for the class name `ConnectionClosedException` might be the best option we have...? We can throw other IOExceptions like before, and skip if the name is that.

Reply:
Checking `ConnectionClosedException` seems good enough to me, because it makes sense that if the connection is already closed then we can omit the error.

Reply:
I think `org.apache.http.ConnectionClosedException` is already in the class path given we have some REST client integration in core, but I might be wrong.

Reply:
Sounds good, I pushed that change

Reply:
That is a different version than being used by the AWS SDK (v5 instead of v4). Also, the AWS bundle shades the library.

Reply:
Another option is we check which HTTP client is being used and change the close behavior based on that.